zenodo/modules/deposit/views_rest.py

```diff
@@ -53,3 +69,176 @@ def language():
         'suggestions': langs
     }
     return jsonify(d)
+
+
+def _mimetype_whitelist():
+    return current_app.config['ZENODO_METAFILE_MIMETYPE_WHITELIST']
+
+
+def validate_metadata_mimetype(f):
+    """Decorator to validate the mimetype."""
+    @wraps(f)
+    def inner(self, *args, **kwargs):
+        mimetype = request.headers.get('Content-Type')
+        if mimetype not in _mimetype_whitelist():
+            abort(400, '"{}" is not an acceptable mimetype.'.format(mimetype))
+        return f(self, *args, **kwargs)
+    return inner
+
+
+read_permission = check_oauth2_scope(
+    lambda record: deposit_read_permission_factory(record=record).can(),
+    write_scope.id, metadata_read_scope.id)
+
+update_permission = check_oauth2_scope(
+    lambda record: deposit_update_permission_factory(record=record).can(),
+    write_scope.id, metadata_write_scope.id)
+
+
+METAFILE_PERMISSIONS = {
+    'read': read_permission,
+    'update': update_permission,
+}
+
+
+def check_deposit_permission(action):
+    """Deposit permission decorator."""
+    def _permission_builder(f):
+        @wraps(f)
+        def decorator(self, record=None, *args, **kwargs):
+            if not METAFILE_PERMISSIONS[action](record=record).can():
+                abort(403)
+            return f(self, record=record, *args, **kwargs)
+        return decorator
+    return _permission_builder
+
+
+def get_or_create_metadata_bucket(record):
+    if not record['_buckets'].get('metadata'):
+        metadata_bucket = Bucket.create(
+            quota_size=current_app.config['ZENODO_BUCKET_QUOTA_SIZE'],#lignusfix maybe define a smaller limit
+            max_file_size=current_app.config['ZENODO_MAX_FILE_SIZE'],
+        )
+        # locked or unlocked?
+        record['_buckets']['metadata'] = str(metadata_bucket.id)
+        record.commit()
+        RecordsBuckets.create(record=record.model, bucket=metadata_bucket)
+    else:
+        metadata_bucket = Bucket.query.get(record['_buckets']['metadata'])
+    return metadata_bucket
+
+
+class DepositMetadataResource(MethodView):
+    """Deposit metadata resource."""
+
+    @validate_metadata_mimetype
```

Comment:
Mostly relevant for later on, when we refine the permissions, but this decorator should be probably moved down the order, since we first want to make sure the user has the permissions to perform the action, and then reveal validation information